src/metabase/models/user_parameter_value.clj

```diff
@@ -0,0 +1,51 @@
+(ns metabase.models.user-parameter-value
+  (:require
+   [metabase.api.common :as api]
+   [metabase.models.interface :as mi]
+   [metabase.util.malli :as mu]
+   [metabase.util.malli.schema :as ms]
+   [methodical.core :as methodical]
+   [toucan2.core :as t2]))
+
+;;; ----------------------------------------------- Entity & Lifecycle -----------------------------------------------
+
+(methodical/defmethod t2/table-name :model/UserParameterValue [_model] :user_parameter_value)
+
+(doto :model/UserParameterValue
+  (derive :metabase/model))
+
+(t2/deftransforms :model/UserParameterValue
+  {:value mi/transform-json})
+
+(mu/defn upsert!
+  "Upsert or delete parameter value set by the user."
+  [user-id      :- ms/PositiveInt
+   parameter-id :- ms/NonBlankString
+   value        #_#_:- [:maybe :Any]]
```

Comment:
:Any isn't a malli schema and `[:maybe :any]` is redundant, since nil is in :any. you can just leave off the `:- ...` bit:
```suggestion
   value]
```

Reply:
Is it worth actually deleting the user parameter value record when `value` is nil? The docstring mentions this can delete the parameter, but [looking here](https://github.com/camsaul/toucan2/blob/master/test/toucan2/update_test.clj) I don't think the call to t2/update! can. Doesn't that just set the value to nil?